src/main/java/org/cryptomator/linux/quickaccess/DolphinPlaces.java

```diff
@@ -0,0 +1,161 @@
+package org.cryptomator.linux.quickaccess;
+
+import org.cryptomator.integrations.common.CheckAvailability;
+import org.cryptomator.integrations.common.DisplayName;
+import org.cryptomator.integrations.common.OperatingSystem;
+import org.cryptomator.integrations.common.Priority;
+import org.cryptomator.integrations.quickaccess.QuickAccessService;
+import org.cryptomator.integrations.quickaccess.QuickAccessServiceException;
+import org.xml.sax.SAXException;
+
+import javax.xml.XMLConstants;
+import javax.xml.transform.Source;
+import javax.xml.transform.stream.StreamSource;
+import javax.xml.validation.SchemaFactory;
+import javax.xml.validation.Validator;
+import java.io.IOException;
+import java.io.StringReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardOpenOption;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+/**
+ * Implemenation of the {@link QuickAccessService} for KDE desktop environments using Dolphin file browser.
+ */
+@DisplayName("KDE Dolphin Places")
+@OperatingSystem(OperatingSystem.Value.LINUX)
+@Priority(90)
+public class DolphinPlaces implements QuickAccessService {
+
+	private static final int MAX_FILE_SIZE = 1 << 15; //xml is quite verbose
+	private static final Path PLACES_FILE = Path.of(System.getProperty("user.home"), ".local/share/user-places.xbel");
+	private static final Path TMP_FILE = PLACES_FILE.resolveSibling("user-places.xbel.cryptomator.tmp");
+	private static final Lock MODIFY_LOCK = new ReentrantReadWriteLock().writeLock();
```

Comment:
If I am not mistaken the lock is never held longer than the runtime of either of the two methods. If your intend is to avoid concurrent modification, simply make both methods synchronized.

Reply:
The `synchronized` keyword only synchonizes within the same class, but the methods are in different classes. That is the reason for using a lock

Reply:
A synchronized _block_ would work, but a lock is fine as well. Why do you use `ReentrantReadWriteLock` over a normal `ReentrantLock`, tho?

Reply:
Oh, right, the Reentrant lock is already exclusive. I changed that.

Reply:
:+1: Resolved by eae20da17fb3a4fc3ec5462ee0845b0fbb1e5f30.